lib/commands/exam.js

```diff
@@ -313,5 +316,35 @@ module.exports = TestCommand.extend({
       case 'json':
         return fs.readJsonSync(file);
     }
+  },
+
+  /**
+   * Adds additional event handlers to config.
+   * 
+   * @param {object} config
+   * @return {object} events
+   */
+  _addBrowserSocketEventsToConfig: function(config) {
+    let events = config.additionalBrowserSocketEvents || {};
+
+    events['set-modules-queue'] = function (modules) {
+      const proto = Object.getPrototypeOf(this)
+      if (!proto.moduleQueue) {
+        proto.moduleQueue = modules;
+      }
+    }
+
+    events['get-next-module'] = function () {
+      const proto = Object.getPrototypeOf(this)
+      const moduleName = proto.moduleQueue.shift();
```

Comment:
You can just use `moduleQueue.shift();` here

Reply:
hm wouldn't moduleQueue be undefined?

Reply:
you may need to use `this.moduleQueue.shift()`